app/src/main/java/org/wikipedia/views/LanguageScrollView.java

```diff
@@ -101,27 +97,39 @@ private void updateTabView(boolean selected, TabLayout.Tab tab) {
         }
     }
 
-    public void setUpLanguageScrollTabData(List<String> languageCodes, Callback callback, int position) {
+    @SuppressWarnings("checkstyle:magicnumber")
+    public void setUpLanguageScrollTabData(@NonNull List<String> languageCodes, int position, @Nullable Callback callback) {
         if (this.callback != null) {
             this.callback = null;
         }
+
+        this.callback = callback;
+        this.languageCodes = languageCodes;
+
         if (horizontalLanguageScroll.getChildCount() > 0) {
             horizontalLanguageScroll.removeAllTabs();
         }
-        this.callback = callback;
-        this.languageCodes = languageCodes;
+
         for (int i = 0; i < languageCodes.size(); i++) {
             TabLayout.Tab tab = horizontalLanguageScroll.newTab();
             tab.setCustomView(createLanguageTab(languageCodes.get(i)));
             horizontalLanguageScroll.addTab(tab);
             updateTabView(false, tab);
         }
-        selectLanguageTab(position);
+
+        if (horizontalLanguageScroll != null && horizontalLanguageScroll.getTabAt(position) != null) {
+            horizontalLanguageScroll.postDelayed(() -> {
```

Comment:
What's the reason for `post()`ing this?  It should work fine without a post.

Reply:
It fixes the issue that the scroll view does not scroll to the selected item after tapping on the last item in the Wikipedia languages screen(the item can not be seen by default).

Add ~5 languages and make sure the last item is not able to see -> when you tapping on the "more" to go to the "Wikipedia languages" -> tap on the last item -> it should scroll to the last item.

Reply:
OK, in that case is there a reason for the 100ms delay? Why not just `post` without delay?

Reply:
Thanks, you're correct. Done!